test/TestServerProjectsLowLevel/dpg-customization/Customization/DPGClient.cs

```diff
@@ -0,0 +1,139 @@
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="mode"/> is null. </exception>
+        public virtual async Task<Response<Product>> GetModelValueAsync(string mode, CancellationToken cancellationToken = default)
+        {
+            RequestContext requestContext = new RequestContext();
+            requestContext.CancellationToken = cancellationToken;
+
+            Response response = await GetModelAsync(mode, requestContext);
+            return Response.FromValue((Product)response, response);
+        }
+
+        /// <summary> Get models that you will either return to end users as a raw body, or with a model added during grow up. </summary>
+        /// <param name="mode"> The mode with which you&apos;ll be handling your returned body. &apos;raw&apos; for just dealing with the raw body, and &apos;model&apos; if you are going to convert the raw body to a customized body before returning to users. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="mode"/> is null. </exception>
+        public virtual Response<Product> GetModelValue(string mode, CancellationToken cancellationToken = default)
+        {
+            RequestContext requestContext = new RequestContext();
+            requestContext.CancellationToken = cancellationToken;
+
+            Response response = GetModel(mode, requestContext);
+            return Response.FromValue((Product)response, response);
+        }
+
+        /// <summary> Post either raw response as a model and pass in &apos;raw&apos; for mode, or grow up your operation to take a model instead, and put in &apos;model&apos; as mode. </summary>
+        /// <param name="mode"> The mode with which you&apos;ll be handling your returned body. &apos;raw&apos; for just dealing with the raw body, and &apos;model&apos; if you are going to convert the raw body to a customized body before returning to users. </param>
+        /// <param name="input"> Please put {&apos;hello&apos;: &apos;world!&apos;}. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="mode"/> or <paramref name="input"/> is null. </exception>
+        public virtual async Task<Response<Product>> PostModelAsync(string mode, Input input, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(input, nameof(input));
+
+            RequestContext requestContext = new RequestContext();
+            requestContext.CancellationToken = cancellationToken;
+
+            var content = new Utf8JsonRequestContent();
+            content.JsonWriter.WriteObjectValue(input);
```

Comment:
Please use the approach Shivangi suggested here as well: https://github.com/Azure/autorest.csharp/pull/1989#discussion_r823034592

Reply:
Yes, it is important to show that ToRequestContent is an internal static method on the model, since this is the approach we will recommend to library authors writing models when DPG GA's.